src/js/renderers/editor-dom.js

```diff
@@ -258,28 +307,31 @@ let destroyHooks = {
   },
 
   [IMAGE_SECTION_TYPE](renderNode, section) {
-    let post = renderNode.parent.postNode;
-    post.sections.remove(section);
-    renderNode.element.parentNode.removeChild(renderNode.element);
+    removeRenderNodeSectionFromParent(renderNode, section);
+    removeRenderNodeElementFromParent(renderNode);
   },
 
   [CARD_TYPE](renderNode, section) {
     if (renderNode.cardNode) {
       renderNode.cardNode.teardown();
     }
-    let post = renderNode.parent.postNode;
-    post.sections.remove(section);
-    renderNode.element.parentNode.removeChild(renderNode.element);
+    removeRenderNodeSectionFromParent(renderNode, section);
+    removeRenderNodeElementFromParent(renderNode);
   }
 };
 
 // removes children from parentNode that are scheduled for removal
 function removeChildren(parentNode) {
   let child = parentNode.childNodes.head;
+  let nextChild, method;
   while (child) {
-    let nextChild = child.next;
+    nextChild = child.next;
```

Comment:
I am not sure if any logic changed here from what it was before; I rearranged some variable declarations in order to throw a more meaningful error ("cannot destroy 'list-item'", e.g.) below on line 332